rl4co/data/utils.py

```diff
@@ -21,3 +25,28 @@ def check_extension(filename, extension=".npz"):
     if os.path.splitext(filename)[1] != extension:
         return filename + extension
     return filename
+
+
+def load_solomon_instance(name, path=None, edge_weights=False):
+    """Load a solomon instance from a file"""
+    if not path:
+        path = "data/solomon/instances/"
+        path = os.path.join(ROOT_PATH, path)
+    file_path = f"{path}{name}.txt"
+    if not os.path.isfile(file_path):
+        vrplib.download_instance(name=name, path=path)
+    return vrplib.read_instance(
+        path=file_path,
+        instance_format="solomon",
+        compute_edge_weights=edge_weights,
+    )
+
+
+def load_solomon_solution(name, path=None):
+    if not path:
+        path = "data/solomon/solutions/"
+        path = os.path.join(ROOT_PATH, path)
+    file_path = f"{path}{name}.sol"
+    if not os.path.isfile(file_path):
+        vrplib.download_solution(name=name, path=path)
```

Comment:
This is the way to download, then can we remove the extra `sol` and `txt` file under data? (we could simply call `os.makedirs` to generate the path for downloading automatically)

Reply:
You are right! We could check if the path exists (which it won't the first time) as additional check before checking for the file itself and just create the path if it doesn't. Do you want me to change it?

Reply:
Heya! Current versions of `vrplib` support this downloading functionality, we consider this somewhat deprecated and plan to remove it at some point (probably for `vrplib` 2.0.0). That should not be a problem since you pin your dependencies, but could be something to keep in mind!

Reply:
Thanks a lot @N-Wouda for your heads up! Do you have any suggestions for alternatives?

Reply:
@fedebotu what we do with PyVRP is that we have our own repository of benchmark instances and best-known solutions, here: https://github.com/PyVRP/Instances. The CVRP and VRPTW instances track those in CVRPLIB, but it's nice to have everything in one place, and one format. We include this as a git submodule in the main PyVRP repo, which makes for easy benchmarking.

For RL4CO that's probably not a great option, since routing isn't all you do. Maybe point users to either our instances, or to CVRPLIB, and ask them to download the files themselves? That could be part of your docs. Alternatively, you could implement the downloading yourself: it is [only a few lines of code](https://github.com/PyVRP/VRPLIB/blob/main/vrplib/download/download_solution.py).

Hope that helps!

Reply:
@N-Wouda thanks a ton for the detailed response! I think, as you suggested, it might be easier to just provide the utils to download in a few lines of code and then include some `vrplib` optional dependency as we are doing now for loading them :)